# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/kbd/kbdrate-nonx86.patch
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- kbd-1.08/src/kbdrate.c	2002-10-11 13:09:01.000000000 +0200
+++ kbd-1.08-fixed/src/kbdrate.c	2003-07-19 11:03:05.000000000 +0200
@@ -237,6 +237,7 @@
 	if(KIOCSRATE_ioctl_ok(rate, delay, silent))	/* sparc? */
 		return 0;
 
+#ifdef __i386__ 
 
 	/* The ioport way */
 
@@ -286,4 +287,8 @@
 			valid_delays[ (value & 0x60) >> 5 ] );
 
 	return 0;
+#else
+	perror( _("No alternative method for this platform") );
+	exit( 1 );
+#endif
 }
